### PR TITLE
sdk/metric: Update package example

### PR DESCRIPTION
**Pull request by pellared**
Main changes:

1. Update comment. The `[package]` comments are not working in "regular" comments.
2. Move "defer cleanup" before setting the global. I think it is a good practice to defer cleanup ASAP.
3. I think there is no need to add reference to resource package.

### CODE REVIEW

File: sdk/metric/example_test.go [hunk 1]
```diff
@@ -58,21 +57,20 @@ func Example() {
...
 	// Handle shutdown properly so that nothing leaks.
 	defer func() {
 		err := meterProvider.Shutdown(context.Background())
 		if err != nil {
 			log.Fatalln(err)
 		}
 	}()
+
+	// Register as global meter provider so that it can
+	// that can used via otel.Meter and accessed using otel.GetMeterProvider.
```
Comment (dmathieu) [inline sdk/metric/example_test.go:69]:
```suggestion
	// then be used via otel.Meter and accessed using otel.GetMeterProvider.
```

File: sdk/metric/example_test.go [hunk 2]
```diff
@@ -58,21 +57,20 @@ func Example() {
...
+	// Most instrumentation libraries use the global meter provider as default.
+	// If the global meter provider is not set then a no-op implementation
+	// is used and which fails to generate data.
```
Comment (dmathieu) [inline sdk/metric/example_test.go:72]:
```suggestion
	// is used, which fails to generate data.
```

File: sdk/metric/example_test.go (hunk 1 above)
Comment (pellared) [reply sdk/metric/example_test.go]: https://github.com/open-telemetry/opentelemetry-go/pull/4540/commits/714dc911e4b96f98fe7b7d61fa16a531c2748541

File: sdk/metric/example_test.go (hunk 2 above)
Comment (pellared) [reply sdk/metric/example_test.go]: https://github.com/open-telemetry/opentelemetry-go/pull/4540/commits/714dc911e4b96f98fe7b7d61fa16a531c2748541

**Review by dmathieu — APPROVED** [review]

**Review by MrAlias — APPROVED** [review]

**Pull request merged**